projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1375,6 +1377,16 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
     @Output()
     public onRowToggle = new EventEmitter<IRowToggleEventArgs>();
 
+    /**
+     * Emitted when the pinned state of a row is changed.
+     * @example
+     * ```html
+     * <igx-grid [data]="employeeData" (onRowPinning)="rowPin($event)" [autoGenerate]="true"></igx-grid>
+     * ```
+     */
+    @Output()
+    public onRowPinning = new EventEmitter<IPinRowEventArgs>();
```

Comment:
If it's an `-ing` event, is it supposed to be cancellable?

Reply:
@damyanpetev Wasn't planned to be cancellable according to the spec, however I see no reason why it can't be. @ChronosSF what do you think?

Reply:
If the corresponding column event is not cancellable this shouldn't be either.

In this case, create a feature-request issue to have them both cancellable separately from this PR